Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -0,0 +1,479 @@
+#include "WarpXOpenPMD.H"
+#include "FieldIO.H"  // for getReversedVec
+
+WarpXOpenPMDPlot::WarpXOpenPMDPlot(bool oneFilePerTS,
+                       const std::string& openPMDFileType)
+  :m_Series(nullptr),
+   m_OneFilePerTS(oneFilePerTS),
+   m_OpenPMDFileType(openPMDFileType)
+{}
+
+WarpXOpenPMDPlot::~WarpXOpenPMDPlot()
+{
+  if (nullptr != m_Series) {
+    m_Series->flush();
+    delete m_Series;
+  }
+}
+
+
+//
+//
+//
+void WarpXOpenPMDPlot::GetFileName(std::string& filename, int ts)
+{
+  std::string dir = "diags/";
+  std::string prefix="/plot";
+
+  // Write openPMD format: only for level 0
+  filename = std::string(dir);
+  filename.append(m_OpenPMDFileType);
+
+  if (m_OneFilePerTS)
+    filename += amrex::Concatenate(prefix, ts);
+  else
+    filename.append(prefix);
+
+  filename += std::string(".") + m_OpenPMDFileType;
+}
+
+
+void WarpXOpenPMDPlot::SetOpenPMDType(const std::string& type)
```

Comment:
same as in the constructor:
```suggestion
void WarpXOpenPMDPlot::SetOpenPMDType(const std::string type)
```
and then `std::move` in the `type` variable :)

Reply:
this function is not used anymore, so. I removed it.